Lib/test/test_typing.py

```diff
@@ -8611,6 +8599,32 @@ class C:
         self.assertEqual(get_type_hints(C, globals())['classvar'], ClassVar[int])
         self.assertEqual(get_type_hints(C, globals())['const'], Final[int])
 
+    def test_special_forms_nesting(self):
```

Comment:
Could you add a comment saying why we're explicitly testing these patterns, even though some of them are quite odd code to write? :)

Reply:
Sure I've added one.